geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/registry/ProtobufOperationContextRegistry.java

```diff
@@ -152,5 +153,11 @@ private void addContexts() {
             new OqlQueryRequestOperationHandler(),
             opsResp -> ClientProtocol.Message.newBuilder().setOqlQueryResponse(opsResp),
             new ResourcePermission(Resource.DATA, Operation.READ)));
+
+    operationContexts.put(MessageTypeCase.KEYSETREQUEST,
+        new ProtobufOperationContext<>(ClientProtocol.Message::getKeySetRequest,
+            new KeySetOperationHandler(),
+            opsResp -> ClientProtocol.Message.newBuilder().setKeySetResponse(opsResp),
+            KeySetOperationHandler::determineRequiredPermission));
```

Comment:
Is it `determineRequiredPermission` because each key may have a different required permission?

Reply:
No, it's because we need to get the name of the region from the request to check our permission.  We require region level access or we'll fail the request (which could theoretically be incorrect if we happen to have key level access for every key without having region level read permission, but I'm dismissing that case as stupid).